caelum-revelat.ts

```
@@ -120,6 +120,31 @@ export class BinaryFilterExpression {
 		}
 	}
 
+	/**
+	 * Cast an un-typed {@linkcode BinaryFilterExpression} as an instance.
+	 *
+	 * Does not throw. Returns `null` if the passed value is invalid.
+	 */
+	static tryFrom(expression: unknown): BinaryFilterExpression | null {
+		if (
+			typeof expression === "object" &&
+			expression &&
+			"key" in expression &&
+			typeof expression.key === "string" &&
+			"operator" in expression &&
+			typeof expression.operator === "string" &&
+			isOperator(expression.operator) &&
+			"value" in expression
+		) {
+			const { key, operator, value } = expression;
+			const not = "not" in expression && !!expression.not;
+
+			return new BinaryFilterExpression({ key, operator, value, not });
+		}
+
+		return null;
+	}
```

Should this return `null` or throw that it couldn't be parsed. The naming `tryFrom` would make me expect that it would throw if it's not a valid filter.

---

If you were to throw instead, you could also provide context on why it failed to parse, rather than something unexpectedly returning `null`.

---

I was trying to avoid throwing in this feature. Probably the wrong approach. I'll keep thinking on it.